### ISSUE TITLE
Bug: [import/no-unresolved] Error when importing `typescript-eslint` in a pnpm setup

### ISSUE THREAD

**Issue by lnuvy** [typescript-eslint/typescript-eslint#10525]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://github.com/lnuvy/eslint9-eslint-typescript-8.18.1

### Repro Code

```TypeScript
import tseslint from "typescript-eslint";
```

### ESLint Config

```javascript
import js from "@eslint/js";
import globals from "globals";
import tseslint from "typescript-eslint"; // An error occurs on this line

export default tseslint.config(
  ...
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strict": true,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "target": "es5",
    "module": "NodeNext",
    "moduleResolution": "nodenext",
    "baseUrl": "."
  }
}
```

### Expected Result

expected the import line to work without errors

### Actual Result

Unable to resolve path to module 'typescript-eslint'.eslint[import/no-unresolved](https://github.com/import-js/eslint-plugin-import/blob/v2.31.0/docs/rules/no-unresolved.md)

### Additional Info

Hello!
this is an issue I discovered while setting up ESLint 9 in a monorepo.
I found a potentially related issue (#10508), but there is a key difference: the error does not occur in the import statement for that case. I apologize if this turns out to be a duplicate that I overlooked.

I'm sharing a reproducible repository and images.

<img width="779" alt="스크린샷 2024-12-20 오전 2 18 06" src="https://github.com/user-attachments/assets/6cb1c048-3f69-451f-892b-153210b7160e" />
<img width="944" alt="스크린샷 2024-12-20 오전 2 48 58" src="https://github.com/user-attachments/assets/ebde6176-6567-472a-a6dd-1332e94a837f" />

First image's project uses the Yarn package manager, while the images showing the error are from a project using the pnpm package manager.

Even after trying various versions of TypeScript, ESLint, and typescript-eslint, the error consistently occurs only in pnpm projects.

Functionally, everything works as expected, so using something like `// eslint-disable-next-line import/no-unresolved` is an option. but, I think this is not a fundamental solution, which is why I am reporting this issue.

Thank you :)

**Comment by kirkwaiblinger** [comment]
Hi @lnuvy ! 

I don't think this is something we can do anything with, at least not in its current incarnation... Granted that you say the code works at runtime, this seems like just an issue with [eslint-plugin-import](https://github.com/import-js/eslint-plugin-import). Have you filed an issue there?

**Comment by bradzacher** [comment]
yup -- nothing to do with our project here!
we don't own `eslint-plugin-import` or any of their infra.
if their plugin cannot resolve our package under pnpm then likely you've misconfigured their plugin or their plugin is missing support.

Please reach out to them for help!

**Comment by lnuvy** [comment]
@kirkwaiblinger @bradzacher 
Thank you for advice!

Since the issue occurred in the import line, I hadn't considered that it might be caused by another plugin. 😢

I'll check the `eslint-plugin-import` community for more information.
